Fix Help Information in the Instrument Viewer (#34039)
mantidproject/mantid: docs/source/workbench/instrumentviewer.rst

Overview
--------

The Mantid visualisation system allows users to view the geometry of an instrument and also select and query individual detectors.
An instrument is always shown in the context of a workspace, which contains data recorded with this instrument.
The detectors are coloured according to the data in the corresponding spectra of the workspace.
The user can select individual detectors by clicking on them and request to see the data recorded by this detector as a table or a graph.
The instrument view also allows users to see close-ups of any component.

Instrument View Pane
--------------------
The instrument view window can be obtained by right clicking on the workspace of interest and selecting ``Show Instrument``.

> A 2D projection of the detector arrangement will be visible with each detector pixel color coded depending on the integrated number of counts in its corresponding spectrum, as in the image below.

Reviewer: A mixture of color and colour is used in this document. Whilst matplotlib uses the American spelling and so specific references to matplotlib features need to be spelled correctly in relation to that I think we need some consistency when using these words outside that context. In this case can we change `detector pixel color` to `detector pixel colour`?

Reviewer: Are we not changing this one?

Reviewer: Ah I see we're going with American spelling throughout so ignore my comment here.

Reviewer 2: Yeah, it'd just end up horrible with the mix of mpl and our own code that uses the US spelling.

.. figure:: ../images/Workbench/InstrumentViewer/Overview.png
    :align: center
    :width: 635

It is possible to alter the view of the instrument by holding down either mouse button and moving (dragging) it within the window.
There are other mouse and key board controls, which are listed at the bottom of the instrument view window.

> The control panel of the instrument view has four tabs: Render, Pick, Mask, and Instrument.

Reviewer: ```suggestion
The control panel of the instrument view has four tabs: ``Render``, ``Pick``, ``Mask``, and ``Instrument``.
```

Render Tab
----------
The Render tab contains controls for managing the on-screen appearance of the instrument and collected data.
The top-most combo-box control allows the user to select the way the instrument is projected onto the screen.
The default setting is ``Full 3D`` which gives a 3D view of the instrument in an orthogonal projection.

> The other options are "unwrapped views" projecting the instrument onto a curved surface and then unwrapping it onto the screen.

Reviewer: Should unwrapped views be formatted differently? e.g. ``unwrapped views``

Reviewer 2: Nah, it's just a turn of phrase here, rather than referencing anything on the GUI. That sentence is poor though, so I'll rewrite it anyway

Reviewer: I think for turns of phrase we generally use single quotes - certainly makes it less confusing that double quotes which potentially could have been used accidentally instead of the code formatting...

Right-clicking resets the zoom back to showing the entire instrument.

.. figure:: ../images/Workbench/InstrumentViewer/RenderTab.png
    :align: center
    :width: 635

The next control is ``Axis View``, visible only in the 3D mode, which resets the view so that the instrument is fully visible and the specified axis is perpendicular to the screen.

The ``Display Settings`` button controls the appearance of the instrument in the view.
The ``Use OpenGL`` option toggles between two display modes of a flat view: one that uses OpenGL to render the instrument and one that doesn't.

.. figure:: ../images/Workbench/InstrumentViewer/DisplaySettings.png
    :align: center
    :width: 635

This option can be useful if the instrument is viewed over a slow network connection, for example.

The ``Save image`` button allows the image to be save into a file.

The colour bar axis below maps the colours of the detectors to the integrated number of counts in their spectra.
The axis also defines the minimum and maximum values, which can be edited using the text boxes below and above the colour bar.

> Mantid comes with a number of color map files and these can be loaded by selecting the ``Display Settings`` -> ``Color Map``.

Reviewer: Not sure which spelling of colour/color to use here.

associated with the clicked detector. The user can keep curves of interest for longer by right clicking on the detector and selecting ``Store curve``
from the context menu.

The context menu also gives the option to save the content of the mini-plot to a new workspace. There is no limit to the number of curves
that can be added to the mini-plot. Curves can be removed by right-clicking on the mini-plot and selecting the line the user wishes to remove. The same
menu allows for switching between linear and log scales for the Y axis.

.. |singlePixel| image:: ../images/PickTabPickPixelButton.png

Tube Selection
##############

The |tubeBank| tool allows the user to click a detector and have the mini-plot display the data from the immediate parent of the
detector. Usually a tube or bank, depending on the instrument.

When using this tool, the user can right-click on the mini-plot to select between a sum of the counts of all detectors in the bank/tube or counts integrated
over the time bins vs the detector position in the tube.

.. |tubeBank| image:: ../images/PickTabPickTubeButton.png

> .. image:: ../images/Workbench/InstrumentViewer/PickTubeContextMenu.png

Reviewer: The image could do with being centred.

Create a Single Crystal Peak List
#################################

The |singleCrystal| tool allows the user to create a single crystal peak list and add a peak to it. First, select the tool and click a detector. Then
click on the peak in the mini-plot. If it is the first peak Workbench will create a :ref:`PeaksWorkspace <PeaksWorkspace>` named ``SingleCrystalPeakTable``
and add a peak to it. Peak markers will then be displayed over the instrument.

.. |singleCrystal| image:: ../images/PickTabAddPeakButton.png

> .. image:: ../images/Workbench/InstrumentViewer/SingleCrystalPeakExample.png

Reviewer: The image could do with being centred

Single Crystal Peak Eraser
##########################

The |eraser| tool can be used to remove single crystal peaks by clicking on or dragging over the peak markers. The hatched rectangle indicates the
area where peaks will be erased. The rows in the :ref:`PeaksWorkspace <PeaksWorkspace>` are then deleted with the markers. The size of the rectangle
can be adjusted by scrolling.

.. |eraser| image:: ../images/PickTabEraseButton.png

Peak Comparison Tool
####################

The |peakComparison| tool can be used to calculate the reciprocal angle between two single crystal peaks. The hatched box indicates the area in which peaks
will be selected. Click on a single peak marker to select it, then again on another marker. There will be a line drawn between the two peaks and the information
box will show the list of peaks falling into this spectrum.

.. |peakComparison| image:: ../images/PickTabCPCompare.png

> .. image:: ../images/Workbench/InstrumentViewer/PeakComparisonExample.png

Reviewer: The image could do with being centred

Peak Alignment Tool
###################

The |peakAlign| tool allows for the calculation of in-plane and out-of-plane angles between a user defined plane and an out-of-plane peak in Q-space. The plane
must be defined first, by selecting two peaks. Clicking while they are within the hatched box will select them, and mark them with a blue circle. Subsequent peaks
will be highlighted by green circles. The information box will contain the in-plane angles (phi) and out-of-plane angles (theta) for each green peak.

Selections can be cleared by right-clicking the instrument view and selecting ``Clear alignment plane``.

.. |peakAlign| image:: ../images/PickTabCPAlign.png

> .. image:: ../images/Workbench/InstrumentViewer/PeaksAlignmentExample.png

Reviewer: The image could do with being centred.

.. |rectangle| image:: ../images/PickTabRectButton.png
.. |ellipseRing| image:: ../images/PickTabElRingButton.png
.. |rectangleRing| image:: ../images/PickTabRectRingButton.png
.. |sector| image:: ../images/PickTabSectorButton.png
.. |arbitrary| image:: ../images/PickTabArbitraryShape.png

The Draw tab contains tools for creating and editing geometrical shapes, which can be used for selecting regions of interest (ROI) or for masking or grouping detectors.
The tab contains a smaller toolbar, a shape property browser, and a set of buttons to create shapes.

.. figure:: ../images/Workbench/InstrumentViewer/DrawTab.png
    :align: center
    :width: 635

Masks are created by using six tools:

- |ellipse| Ellipse
- |rectangle| Rectangle
- |ellipseRing| Ellipse Ring
- |rectangleRing| Rectangle Ring
- |sector| Sector
- |arbitrary| Brush: This allows for drawing of masks arbitrarily. The size of the brush can be changed by scrolling.

There is no limit on the number of shapes. The shapes can be selected and edited by selecting |edit|.
Click on the shaded part of a shape and drag to move it around or move a control point to resize.

.. |edit| image:: ../images/PickTabEditButton.png

> .. image:: ../images/Workbench/InstrumentViewer/DrawShapesExample.jpg

Reviewer: This image could do with being centred. It could also do with some space between the text and the image.

The property browser displays the parameters of the selected shape. All sizes are given in the coordinates of the unwrapped surface the shapes are drawn on.

> For example, for a spherical surface it will be polar and azimuth angles in radians. The shapes are "attached" to the surface and move with it when the view is zoomed or moved.

Reviewer: ```suggestion
For example, for a spherical surface it will be polar and azimuth angles in radians. The shapes are 'attached' to the surface and move with it when the view is zoomed or moved.
```

Reviewer 2: Why does this need to be changed?

Reviewer: I think single quotes for turn of phrase as discussed above.

Reviewer 2: Is that a standard? I'd use double quotes for that kind of thing like I'd do with my fingers when speaking

Reviewer: I guess its a personal preference thing so lets leave it as it is.

|zoom| can be used to zoom in and out from the render.

.. |zoom| image:: ../images/PickTabZoomButton.png

> .. image:: ../images/Workbench/InstrumentViewer/DrawPropertyBrowser.png

Reviewer: Can this be centred too?

- To apply the shapes to the workspace press the ``Apply to Data`` button. The shapes will be applied into the underlying workspace and shown in the instrument display.
- To see the effect of applying the changes, but without affecting the underlying workspace, click the ``Apply to View`` button.
  This will change the view you are looking at, but not the underlying data in the workspace. Changes applied in this way can be reverted using ``ClearAll`` button.

Masks and Regions of Interest (ROI's) can also be saved into a separate workspace or into a file, using the options in the ``Apply and Save`` drop-down.

If the ``Group`` option is selected, the selected detectors can be extracted or summed into a workspace using the ``Save`` drop-down.

Bin Masking
###########

When the ``Mask`` option is selected and the integration range is reduced, only bins within the selected range will be masked instead of whole detectors.
The shapes created with the drawing tools now select the detectors for which the bins will be masked.

Clicking the ``Apply bin mask to view`` button will use the :ref:`MaskBins <algm-MaskBins>` algorithm mask the selected bins in the data workspace.

> .. image:: ../images/Workbench/InstrumentViewer/DrawBinMask.png

Reviewer: Could we have some space around the image? Feels crowded by the text.

The ``Apply to Data`` button will apply all defined detector and bin masks using the relevant algorithms.

Instrument Tab
--------------

This contains a list of components and component assemblies as they are named and arranged in the :ref:`Instrument Definition File <InstrumentDefinitionFile>`.
For example, in some instruments the detectors are arranged into banks, in others - a long list.
Clicking on a component moves the view of the instrument to centre on that component.

Python Control
--------------
Many aspects of the instrument view can be controlled from Python.

> To use the Python interface for the InstrumentViewer, use the following code to import the necessary libraries:

Reviewer: ```suggestion
To use the Python interface for the Instrument Viewer, use the following code to import the necessary libraries:
```

    myiv.reset_view()
    sys.exit(app.exec_())

> NOTE: If the Instrument Viewer is launched outside outside of Workbench on RHEL_7, the user will need to click on the viewing widget to allow it to render correctly.

Reviewer: ```suggestion
NOTE: If the Instrument Viewer is launched outside of Workbench on RHEL_7, the user will need to click on the viewing widget to allow it to render correctly.
```
